convert/plugin_updates.go

```diff
@@ -94,3 +99,90 @@ func autoGenerateNamespaceForRLAPluginConsumerGroups(plugin *kong.ConsumerGroupP
 	}
 	return nil
 }
+
+func updatePlugins(content *file.Content) {
+	for idx := range content.Plugins {
+		plugin := &content.Plugins[idx]
+		updateLegacyPluginConfig(plugin)
+	}
+
+	for _, service := range content.Services {
+		for _, plugin := range service.Plugins {
+			updateLegacyPluginConfig(plugin)
+		}
+	}
+
+	for _, route := range content.Routes {
+		for _, plugin := range route.Plugins {
+			updateLegacyPluginConfig(plugin)
+		}
+	}
+
+	for _, consumer := range content.Consumers {
+		for _, plugin := range consumer.Plugins {
+			updateLegacyPluginConfig(plugin)
+		}
+	}
+
+	for _, consumerGroup := range content.ConsumerGroups {
+		for _, plugin := range consumerGroup.Plugins {
+			updateLegacyPluginConfig(&file.FPlugin{
+				Plugin: kong.Plugin{
+					ID:     plugin.ID,
+					Name:   plugin.Name,
+					Config: plugin.Config,
+				},
+			})
+		}
+	}
+}
+
+func updateLegacyPluginConfig(plugin *file.FPlugin) {
+	if plugin != nil && plugin.Config != nil {
+		config := plugin.Config.DeepCopy()
+
+		config = updateLegacyFieldToNewField(config, "blacklist", "deny", "")
+
+		config = updateLegacyFieldToNewField(config, "whitelist", "allow", "")
+
+		if plugin.Name != nil {
+			pluginName := *plugin.Name
+			if pluginName == awsLambdaPluginName {
+				config = removeDeprecatedFields3x(config, "proxy_scheme", pluginName)
+			}
+			if pluginName == prefunctionPluginName || pluginName == postfunctionPluginName {
+				config = updateLegacyFieldToNewField(config, "functions", "access", pluginName)
+			}
+		}
+
+		plugin.Config = config
+	}
+}
+
+func updateLegacyFieldToNewField(pluginConfig kong.Configuration,
+	oldField, newField, pluginName string,
+) kong.Configuration {
+	var changes int
+	if _, ok := pluginConfig[oldField]; ok {
+		pluginConfig[newField] = pluginConfig[oldField]
+		delete(pluginConfig, oldField)
+		changes++
+	}
+	if changes > 0 {
+		cprint.UpdatePrintf("Automatically converted legacy configuration field \"%s\" to the new field %s in plugin %s\n",
+			oldField, newField, pluginName)
+	}
+	return pluginConfig
+}
+
+func removeDeprecatedFields3x(pluginConfig kong.Configuration, fieldName, pluginName string) kong.Configuration {
+	var changes int
```

Comment:
If removing one field at a time, this function can be simplified, no? `changes` and the if condition based on it can be removed?